core/domain/email_manager_test.py

```diff
@@ -88,6 +90,291 @@ def test_sender_id_validation(self):
+
+            messages = self.mail_stub.get_sent_messages(to=self.NEW_USER_EMAIL)
+            self.assertEqual(len(messages), 0)
+
+    def test_role_emails_sent_are_correct(self):
+        with self.can_send_emails_ctx, self.can_send_editor_role_email_ctx:
+            email_manager.send_role_notification_email(
+                self.editor_id, self.new_user_id, rights_manager.ROLE_VIEWER,
+                self.exploration.id, self.exploration.title)
+
+            messages = self.mail_stub.get_sent_messages(to=self.NEW_USER_EMAIL)
+            self.assertEqual(len(messages), 1)
+
+            all_models = email_models.SentEmailModel.get_all().fetch()
+            self.assertEqual(len(all_models), 1)
+
+            sent_email_model = all_models[0]
+
+            # Check that email details are correct.
+            self.assertEqual(
+                sent_email_model.recipient_id,
+                self.new_user_id)
+            self.assertEqual(
+                sent_email_model.recipient_email, self.NEW_USER_EMAIL)
+            self.assertEqual(
+                sent_email_model.sender_id, feconf.SYSTEM_COMMITTER_ID)
+            self.assertEqual(
+                sent_email_model.sender_email,
+                'Site Admin <%s>' % feconf.SYSTEM_EMAIL_ADDRESS)
+            self.assertEqual(
+                sent_email_model.intent,
+                feconf.EMAIL_INTENT_EDITOR_ROLE_NOTIFICATION)
+            self.assertEqual(
+                sent_email_model.subject,
+                self.expected_email_subject)
+
+    def test_correct_rights_are_written_in_manager_role_email_body(self):
+        expected_email_html_body = (
+            'Hi newuser,<br>'
+            '<br>'
+            '<b>editor</b> has granted you Manager rights to their '
+            'learning exploration, '
+            '<a href="http://www.oppia.org/A">"Title"</a>'
+            ', on Oppia.org.<br>'
+            '<br>'
+            'This allows you to:<br>'
+            '<ul>'
+            '<li>Change the exploration permissions</li><br>'
+            '<li>Edit the exploration</li><br>'
+            '<li>View and playtest the exploration</li><br>'
+            '</ul><br>'
+            'You can find exploration '
+            '<a href="http://www.oppia.org/A">here</a>.<br>'
+            '<br>'
+            'Thanks, and happy collaborating!<br>'
+            '<br>'
+            'Best Wishes,<br>'
+            'The Oppia Team<br>'
+            '<br>'
+            'You can change your email preferences via the '
+            '<a href="https://www.example.com">Preferences</a> page.'
+        )
+
+        expected_email_text_body = (
+            'Hi newuser,\n'
+            '\n'
+            'editor has granted you Manager rights to their '
+            'learning exploration, '
+            '"Title"'
+            ', on Oppia.org.\n'
+            '\n'
+            'This allows you to:\n'
+            'Change the exploration permissions\n'
```

Comment:
Yes and yes (that is indeed what I meant).

On Tuesday, April 19, 2016, Prasanna Patil <notifications@github.com> wrote:

> In core/domain/email_manager_test.py
> <https://github.com/oppia/oppia/pull/1737#discussion_r60357436>:
>
> > +            'The Oppia Team<br>'
> > +            '<br>'
> > +            'You can change your email preferences via the '
> > +            '<a href="https://www.example.com">Preferences</a> page.'
> > +        )
> > +
> > +        expected_email_text_body = (
> > +            'Hi newuser,\n'
> > +            '\n'
> > +            'editor has granted you Manager rights to their '
> > +            'learning exploration, '
> > +            '"Title"'
> > +            ', on Oppia.org.\n'
> > +            '\n'
> > +            'This allows you to:\n'
> > +            'Change the exploration permissions\n'
>
> Oh I see, you mean to say we should file a new issue so that <li>list
> point</li> auto matically gets converted to '- list point' (a hyphen and
> a space) ??
> If that is right then I will file the new issue. Meanwhile I suggest we
> keep the plain text tests intact and they should be modified as a part of
> this new issue ?
> What do you think ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/oppia/oppia/pull/1737/files/dfbfc65d81faa3c137ded81f30169f3a674fb9b9#r60357436>
>